ray-operator/controllers/utils/util.go

```diff
@@ -165,3 +166,64 @@ func FilterContainerByName(containers []corev1.Container, name string) (corev1.C
 
 	return corev1.Container{}, fmt.Errorf("can not find container %s", name)
 }
+
+// GetHeadGroupServiceAccountName returns the head group service account if it exists.
+// Otherwise, it returns the name of the cluster itself.
+func GetHeadGroupServiceAccountName(cluster *rayiov1alpha1.RayCluster) string {
+	headGroupServiceAccountName := cluster.Spec.HeadGroupSpec.Template.Spec.ServiceAccountName
+	if headGroupServiceAccountName != "" {
+		return headGroupServiceAccountName
+	}
+	return cluster.Name
+}
+
+func TestGetHeadGroupServiceAccountName(t *testing.T) {
```

Comment:
Let's move test to https://github.com/ray-project/kuberay/blob/master/ray-operator/controllers/utils/util_test.go?